NPE with H2 v1.4.200 repeatable read select queries

Looks like switching to repeatable read isolation level breaks `SELECT` queries.

Discovered while trying to use sqlline which automatically switches to repeatable read (https://github.com/julianhyde/sqlline/issues/364).

Reproduction steps here: [repro.txt](https://github.com/h2database/h2database/files/3748580/repro.txt)

Trace here:
[test.trace.db.txt](https://github.com/h2database/h2database/files/3748581/test.trace.db.txt)

Stacktrace:
```
org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.NullPointerException" [50000-200]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:505)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:429)
        at org.h2.message.DbException.get(DbException.java:194)
        at org.h2.message.DbException.convert(DbException.java:347)
        at org.h2.message.DbException.toSQLException(DbException.java:319)
        at org.h2.message.TraceObject.logAndConvert(TraceObject.java:366)
        at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:203)
        at org.h2.tools.Shell.execute(Shell.java:470)
        at org.h2.tools.Shell.promptLoop(Shell.java:295)
        at org.h2.tools.Shell.runTool(Shell.java:152)
        at org.h2.tools.Shell.main(Shell.java:81)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at coursier.cli.launch.Launch$.$anonfun$launch$6(Launch.scala:116)
        at coursier.cli.launch.Launch$.$anonfun$task$28(Launch.scala:295)
        at coursier.cli.launch.Launch$.run(Launch.scala:356)
        at coursier.cli.Coursier$.$anonfun$runA$2(Coursier.scala:79)
        at coursier.cli.Coursier$.$anonfun$runA$2$adapted(Coursier.scala:67)
        at coursier.cli.CommandAppPreA.run(CommandAppPreA.scala:22)
        at caseapp.core.app.CommandAppWithPreCommand.$anonfun$main$5(CommandAppWithPreCommand.scala:98)
        at caseapp.core.app.CommandAppWithPreCommand.$anonfun$main$5$adapted(CommandAppWithPreCommand.scala:98)
        at scala.util.Either.fold(Either.scala:191)
        at caseapp.core.app.CommandAppWithPreCommand.$anonfun$main$3(CommandAppWithPreCommand.scala:98)
        at caseapp.core.app.CommandAppWithPreCommand.$anonfun$main$3$adapted(CommandAppWithPreCommand.scala:84)
        at scala.Option.foreach(Option.scala:407)
        at caseapp.core.app.CommandAppWithPreCommand.main(CommandAppWithPreCommand.scala:84)
        at coursier.cli.Coursier.main(Coursier.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at coursier.bootstrap.launcher.a.a(Unknown Source)
        at coursier.bootstrap.launcher.Launcher.main(Unknown Source)
Caused by: java.lang.NullPointerException
        at org.h2.engine.Session.addTableToDependencies(Session.java:1874)
        at org.h2.engine.Session.startStatementWithinTransaction(Session.java:1832)
        at org.h2.command.Command.executeQuery(Command.java:190)
        at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:222)
        at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:201)
        ... 28 more
```

--- a/h2/src/docsrc/html/changelog.html
+++ b/h2/src/docsrc/html/changelog.html
@@ -21,6 +21,8 @@ Change Log
 
 <h2>Next Version (unreleased)</h2>
 <ul>
+<li>Issue #2191: NPE with H2 v1.4.200 repeatable read select queries
+</li>
 <li>Issue #1390: Add standard-compliant ARRAY data type syntax
 </li>
 <li>PR #2186: Refactor Parser.parseColumnWithType() and fix some minor issues with CAST

--- a/h2/src/main/org/h2/engine/Session.java
+++ b/h2/src/main/org/h2/engine/Session.java
@@ -1868,10 +1868,13 @@ public class Session extends SessionWithState implements TransactionStore.Rollba
             return;
         }
         addTableToDependencies(table, maps);
-        for (Constraint constraint : table.getConstraints()) {
-            Table ref = constraint.getTable();
-            if (ref != table && ref instanceof MVTable) {
-                addTableToDependencies((MVTable) ref, maps, processed);
+        ArrayList<Constraint> constraints = table.getConstraints();
+        if (constraints != null) {
+            for (Constraint constraint : constraints) {
+                Table ref = constraint.getTable();
+                if (ref != table && ref instanceof MVTable) {
+                    addTableToDependencies((MVTable) ref, maps, processed);
+                }
             }
         }
     }

--- a/h2/src/test/org/h2/test/db/TestTransaction.java
+++ b/h2/src/test/org/h2/test/db/TestTransaction.java
@@ -1025,6 +1025,11 @@ public class TestTransaction extends TestDb {
                 conn2.setAutoCommit(false);
                 Statement stat1 = conn1.createStatement();
                 Statement stat2 = conn2.createStatement();
+                // Test a table without constraints
+                stat1.execute("CREATE TABLE TEST(VALUE INT)");
+                stat1.executeQuery("TABLE TEST").close();
+                stat1.execute("DROP TABLE TEST");
+                // Other tests
                 stat1.execute("CREATE TABLE TEST(ID VARCHAR PRIMARY KEY, VALUE INT)");
                 stat1.execute("INSERT INTO TEST VALUES ('1', 1)");
                 conn1.commit();

